### PR TITLE
Make wolfHSM pass wolfcrypt Test and Benchmark

### PR DESCRIPTION
**Pull request by billphipps**
Initial set of fixes and debug output to get wolfHSM client to pass wolfCrypt Test and Benchmark.  So far:
1. Added debug cryptocb handling
2. AES-CBC: update IV after operation.

WIP for additional fixes.  Maybe do those in later PR's to keep things tight?

### CODE REVIEW

File: src/wh_client_cryptocb.c [hunk 1]
```diff
@@ -75,51 +101,94 @@ int wh_Client_CryptoCb(int devId, wc_CryptoInfo* info, void* inCtx)
...
-
+            uint16_t blocks = info->cipher.aescbc.sz / AES_BLOCK_SIZE;
+            size_t last_offset = (blocks - 1) * AES_BLOCK_SIZE;
+            uint8_t* ciphertext = NULL;
+            dataSz =    sizeof(packet->cipherAesCbcReq) +
+                        info->cipher.aescbc.aes->keylen +
+                        AES_IV_SIZE +
+                        info->cipher.aescbc.sz;
+
+            /* III 0 size check is done in wolfCrypt */
```
Comment (bigbrett) [inline src/wh_client_cryptocb.c:112]: whats with the `III`?

File: src/wh_client_cryptocb.c [hunk 2]
```diff
@@ -230,39 +301,91 @@ int wh_Client_CryptoCb(int devId, wc_CryptoInfo* info, void* inCtx)
                 WC_ALGO_TYPE_PK,
                 WH_PACKET_STUB_SIZE + sizeof(packet->pkRsakgReq),
                 (uint8_t*)packet);
+            printf("RSA KeyGen Req sent:size:%u, e:%u, ret:%d\n",
+                    packet->pkRsakgReq.size, packet->pkRsakgReq.e, ret);
```
Comment (bigbrett) [inline src/wh_client_cryptocb.c:305]: should be guarded by  `DEBUG_CRYPTOCB_VERBOSE`

File: src/wh_client_cryptocb.c [hunk 3]
```diff
@@ -230,39 +301,91 @@ int wh_Client_CryptoCb(int devId, wc_CryptoInfo* info, void* inCtx)
...
             if (ret == 0) {
                 do {
                     ret = wh_Client_RecvResponse(ctx, &group, &action, &dataSz,
                         (uint8_t*)packet);
                 } while (ret == WH_ERROR_NOTREADY);
             }
+            printf("RSA KeyGen Res recv:keyid:%u, rc:%d, ret:%d\n",
+                    packet->pkRsakgRes.keyId, packet->rc, ret);
```
Comment (bigbrett) [inline src/wh_client_cryptocb.c:313]: should be guarded by  `DEBUG_CRYPTOCB_VERBOSE`

File: test/wolfhsm_cfg.h [hunk 4]
```diff
@@ -30,12 +30,12 @@
...
 /* #define WOLFHSM_CFG_SHE_EXTENSION */
 
-#define WOLFHSM_CFG_COMM_DATA_LEN 1152
+#define WOLFHSM_CFG_COMM_DATA_LEN 1280
 
 
 #define WOLFHSM_CFG_NVM_OBJECT_COUNT 30
 #define WOLFHSM_CFG_SERVER_KEYCACHE_COUNT 10
-#define WOLFHSM_CFG_SERVER_KEYCACHE_SIZE 1024
+#define WOLFHSM_CFG_SERVER_KEYCACHE_SIZE 1280
```
Comment (bigbrett) [inline test/wolfhsm_cfg.h:38]: perhaps explicitly set to `WOLFHSM_CFG_COMM_DATA_LEN`?

**Review by bigbrett — APPROVED** [review]

**Review by bigbrett — CHANGES_REQUESTED** [review]
oops, meant to request changes in my prior review - only critical thing is the unguarded printf

File: src/wh_client_cryptocb.c (hunk 1 above)
Comment (billphipps) [reply src/wh_client_cryptocb.c:112]: That's my note format:

```
III means this is informational only
WWW means there's dragons nearby, not necessarily a bug but likely fragile code
XXX means that there is certainly a bug or busted feature.
```

File: test/wolfhsm_cfg.h (hunk 4 above)
Comment (billphipps) [reply test/wolfhsm_cfg.h:38]: Well, the magic number is more like 1190-ish plus some headers and junk to support passing an RSA4096 DER key.  So, we should probably set this to an actual number and then make sure COMM_DATA_LEN is larger than that plus the packet headers.  Ok to leave this way for now.

File: src/wh_client_cryptocb.c (hunk 2 above)
Comment (billphipps) [reply src/wh_client_cryptocb.c:305]: My bad.  Likely to remove all this debug prints.  Should left the PR in draft form. Let's get this one done as is as i'm moving on to ECC.

**Review by bigbrett — APPROVED** [review]
Looks great. @jpbland1 please review and merge if good

**Review by jpbland1 — APPROVED** [review]
Looks good

**Pull request merged**